CHANGELOG.md

```diff
@@ -1,5 +1,58 @@
+
+### Enhancements
+
+#### :gear: Select Tests and Fixtures to Run by Their Metadata ([#2527](https://github.com/DevExpress/testcafe/issues/2527)) by [@NickCis](https://github.com/NickCis)
+
+You can now run only those tests or fixtures whose [metadata](https://devexpress.github.io/testcafe/documentation/test-api/test-code-structure.html#specifying-testing-metadata) contains a specific set of values.
+
+Use the [--test-meta](https://devexpress.github.io/testcafe/documentation/using-testcafe/command-line-interface.html#--test-meta-keyvaluekey2value2) flag to specify values to look for in test metadata.
+
+```sh
+testcafe chrome my-tests --test-meta device=mobile,env=production
+```
+
+To select fixtures by their metadata, use the [--fixture-meta](https://devexpress.github.io/testcafe/documentation/using-testcafe/command-line-interface.html#--fixture-meta-keyvaluekey2value2) flag.
+
+```sh
+testcafe chrome my-tests --fixture-meta subsystem=payments,type=regression
+```
+
+In the API, test and fixture metadata is now passed to the [runner.filter](https://devexpress.github.io/testcafe/documentation/using-testcafe/programming-interface/runner.html#filter) method in the `testMeta` and `fixtureMeta` parameters. Use this metadata to decide whether to run the current test.
+
+```js
+runner.filter((testName, fixtureName, fixturePath, testMeta, fixtureMeta) => {
+    return testMeta.mobile === 'true' &&
+        fixtureMeta.env === 'staging';
+});
+```
+
+#### :gear: Run Dynamically Loaded Tests ([#2074](https://github.com/DevExpress/testcafe/issues/2074))
+
+You can now run tests imported from external libraries or generated dynamically even if the `.js` file you provide to TestCafe does not contain any tests.
+
+Previously, this was not possible because TestCafe required test files to contain the [fixture](https://devexpress.github.io/testcafe/documentation/test-api/test-code-structure.html#fixtures) and [test](https://devexpress.github.io/testcafe/documentation/test-api/test-code-structure.html#tests) directives. Now you can bypass this check. To do this, provide the [--disable-test-syntax-validation](https://devexpress.github.io/testcafe/documentation/using-testcafe/command-line-interface.html#--disable-test-syntax-validation) command line flag.
+
+```sh
+testcafe safari test.js --disable-test-syntax-validation
+```
+
+In the API, use the [disableTestSyntaxValidation](https://devexpress.github.io/testcafe/documentation/using-testcafe/programming-interface/runner.html#run) option.
+
+```js
+runner.run({ disableTestSyntaxValidation: true })
+```
+
+### Bug Fixes
+
+* The `touchstart` event now correctly records touches ([#2856](https://github.com/DevExpress/testcafe/issues/2856))
+* Google Chrome now closes correctly after tests are finished ([#2860](https://github.com/DevExpress/testcafe/issues/2860))
+* TestCafe no longer throws errors in the console ([testcafe-hammerhead/#1769](https://github.com/DevExpress/testcafe-hammerhead/issues/1769))
```

Comment:
Probably you want to say TestCafe no longer throws errors in the following cases:...? 
Or `Fixed mechanism of filtration the internal attributes` / `Fixed testcafe-hammerhead/testcafe proxy errors`?
I mean, if i were a customer I'd take this message literally, 
`TestCafe no longer throws errors in the console` - What does it mean, does TestCafe never throws an error in console?

/cc @miherlosev

Reply:
Internal attributes and node changes now does not provoke MutationObserver notifications.

Reply:
Yeah, I forgot, I wanted to ask about this.